src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -636,6 +648,102 @@ internal CompletionItem ResolveCompletionItem(CompletionItem completionItem)
         }
 
         /// <summary>
+        /// Get definition for a selected sql object using SMO Scripting
+        /// </summary>
+        /// <param name="textDocumentPosition"></param>
+        /// <param name="scriptFile"></param>
+        /// <param name="connInfo"></param>
+        /// <returns> Location with the URI of the script file</returns>
+        internal Location[] GetDefinition(TextDocumentPosition textDocumentPosition, ScriptFile scriptFile, ConnectionInfo connInfo)
+        {
+            // Parse sql
+            ScriptParseInfo scriptParseInfo = GetScriptParseInfo(textDocumentPosition.TextDocument.Uri);
+            if (scriptParseInfo != null && RequiresReparse(scriptParseInfo, scriptFile))
+            {
+                ParseAndBind(scriptFile, connInfo);
+            }
+
+            // Return if parse failed
+            if (scriptParseInfo == null)
+            {
+                return null;
+            }
+
+            // Get token from selected text
+            Token selectedToken = GetToken(scriptParseInfo, textDocumentPosition.Position.Line + 1, textDocumentPosition.Position.Character);
+            if (selectedToken == null)
+            {
+                return null;
+            }
+            // Strip "[" and "]"(if present) from the token text to enable matching with the suggestions.
+            // The suggestion title does not contain any sql punctuation
+            string tokenText = selectedToken.Text.Replace("]",String.Empty).Replace("[",String.Empty);
```

Comment:
We need to handle escaped [ ] characters. It's possible to add a table called [Strange\[Name] for example. Please put this in a Utils file since we'll probably need to handle cases like this in other places.